### PR TITLE
cam6_3_156: Make use of lapack library matrix solve routine in WACCM electric field module

### LINKED ISSUE #999: Remove Numerical Recipes code needed for matrix solve in electric field module
### Issue Type

Code Clean-up

### Issue Description

It has been pointed out that routines from "Numerical Recipes", by Press et al., should not be included in released code due to copyright license restrictions.  WACCM physics efield module makes use of module sv_decomp which contains copies of Numerical Recipes routines to solve a matrix equation.  The matrix equation can be solved via lapack library DGESV routine.

### Will this change answers?

No

### Will you be implementing this yourself?

Yes

### PR DESCRIPTION
**Pull request by fvitt**
This uses the supported lapack library DGESV routine to solve a matrix equation in WACCM physics efield module.

Closes #999

This also includes a fix for a waccmx use case file and a minor change to `apex` module needed for when NAG compiler '-nan' option is used.

### CODE REVIEW

**Review by cacraigucar — APPROVED** [review]

**Review by nusbaume — APPROVED** [review]
Looks good!  I just have two small, non-required change requests.

File: src/chemistry/mozart/mo_jshort.F90 [hunk 1]
```diff
@@ -1601,9 +1604,9 @@ subroutine effxs( x, t, xs )
 !     method:
 !     ln(xs) = A(X)[T-220]+B(X)
 !     X = log of slant column of O2
-!     A,B calculated from chebyshev polynomial coeffs
-!     AC and BC using NR routine chebev.  Assume interval
-!     is 38<ln(NO2)<56.
+!     A,B are calculated from Chebyshev polynomial coeffs
+!     AC and BC using Clenshaw summation algorithm within
+!     the interval is 38<ln(NO2)<56.
```
[the remark this answers is not in the document]
Comment (fvitt) [reply src/chemistry/mozart/mo_jshort.F90]: changed as suggested

File: src/chemistry/mozart/mo_jshort.F90 [hunk 2]
```diff
@@ -1652,56 +1653,50 @@ subroutine calc_params( x, a, b )
...
 
-      function jchebev( a, b, c, m, x )
-!-------------------------------------------------------------
-!     Chebyshev evaluation algorithm
-!     See Numerical recipes p193
-!-------------------------------------------------------------
+      ! Use Clenshaw summation algorithm to evaluate Chebyshev polynomial at point
+      ! [pnt - (o2xs_ulimit + o2xs_llimit)/2]/[(o2xs_ulimit - o2xs_llimit)/2]
+      ! given coefficients coefs within limits lim1 and lim2
+      function evalchebpoly( coefs, pnt ) result(cval)
```
[the remark this answers is not in the document]
Comment (fvitt) [reply src/chemistry/mozart/mo_jshort.F90]: changed as suggested

**Comment by cacraigucar** [comment]
I just realized - is this tag answer changing for CAM%DEV FLT runs?  If so, we should postpone it.

**Comment by fvitt** [comment]
@cacraigucar I don't mind postponing.  However, this PR is bit-for-bit unchanged for all test configurations, including CAM%DEV FLT

**Comment by cacraigucar** [comment]
@fvitt - Since it is non-answer changing, please feel free to proceed.  I just had a momentary panic that it could be answer changing, which we wouldn't want right now.  Again, feel free to move ahead with tagging this.

**Pull request merged**